packages/react-sdk/docusaurus/docs/React/03-ui-components/core/participant-view.mdx

```diff
@@ -3,8 +3,130 @@ id: participant-view
+
+:::note
+
+Please note that we suggest not to create a custom `ParticipantView` component as it contains complex, low-level logic related to performance. Please refer to the [Customization section](#customization) to see how you can tailor the component to your needs.
 
 :::
+
+## General usage
+
+You can use the `ParticipantView` component to create your own call layout:
+
+```tsx
+import '@stream-io/video-react-sdk/dist/css/styles.css';
+import {
+  CallControls,
+  ParticipantView,
+  StreamCall,
+  useParticipants,
+} from '@stream-io/video-react-sdk';
+
+const CallUI = () => {
+  const participants = useParticipants();
+
+  return (
+    <StreamCall
+      callType="default"
+      callId="my-first-call"
+      data={{ create: true }}
+      autoJoin
+    >
+      {participants.map((p) => (
+        // highlight-next-line
+        <ParticipantView participant={p} />
+      ))}
+      <CallControls />
+    </StreamCall>
+  );
+};
+```
+
+## Props
+
+### `participant`
+
+| Type                                                                                                            |
+| --------------------------------------------------------------------------------------------------------------- |
+| [`StreamVideoParticipant`](https://github.com/GetStream/stream-video-js/blob/main/packages/client/src/types.ts) |
+
+The participant whose video/audio feed we want to show.
+
+### `ParticipantViewUI`
+
+| Type                                        |
+| ------------------------------------------- |
+| `ComponentType` \| `ReactElement` \| `null` |
+
+Override the default UI for rendering participant information/actions.
+
+### `VideoPlaceholder`
+
+| Type                                        |
+| ------------------------------------------- |
+| `ComponentType` \| `ReactElement` \| `null` |
+
+Override the default UI that's visible when a participant turned off their video.
+
+### `sinkId`
+
+| Type                    |
+| ----------------------- |
+| `string` \| `undefined` |
+
+In [supported browsers](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/sinkId), this sets the default audio output.
+The value of this prop should be a valid audio output device ID.
+You can set this using `audioOutputDeviceId` field of the [local participant](../../../call-engine/hooks-and-contexts/#uselocalparticipant).
+
+### `videoKind`
+
+| Type                |
+| ------------------- |
+| `video` \| `screen` |
+
+The kind of video stream to play for the given participant.
+
+### `muteAudio`
+
+| Type      |
+| --------- |
+| `boolean` |
+
+You can mute the audio of the given participant (this is a local action, it won't have any effect on the published audio of the participant). The `ParticipantView` will mute the audio of the local participant by default.
+
+### `className`
+
+| Type                    |
+| ----------------------- |
+| `string` \| `undefined` |
+
+Custom class applied to the root DOM element.
+
+### `refs`
```

Comment:
I believe that this prop would definitely deserve more information about why and how.

Reply:
These are currently used only in the `egress-composite` application to notify egress that the "application" is ready. Outside of that I haven't really found a use for these - maybe for the actual web picture in picture API.

But the main point is that the component is "isolated" and it'd be hard to access these elements without it.

Reply:
Added some extra explanation (mentioned that they can use this if they need to add custom event handler to these elements)